kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/batch/KCBQThreadPoolExecutor.java

```diff
@@ -106,4 +105,16 @@ private static String createErrorString(Collection<Throwable> errors) {
                         .collect(Collectors.toList()));
     return String.join(", ", exceptionTypeStrings);
   }
+
+  /**
+   * Checks for BigQuery errors. No-op if there isn't any error.
+   *
+   * @throws BigQueryConnectException if there have been any unrecoverable errors when writing to BigQuery.
+   */
+  public void checkForErrors() throws BigQueryConnectException {
```

Comment:
maybe call this method `errorCount()` that returns the number of unrecoverable errors? and have the calling method throw an exception? right now, the signature seems a bit "bundled", where I am not really checking for anything. it blows up on touching, and there is no way to just "check", if that makes sense.

Reply:
I think an advantage of this design over a simple `errorCount` method is the ability to specify exactly what errors occurred when trying to write to BigQuery.

Some alternatives I can think of to make the API here clearer:

- Rename method to something like `throwIfFailed`, `maybeFail`, `maybeThrowWriteFails`, etc.
- Change method signature to either a `Throwable` that can be `null` if no writes have failed, or an `Optional<Throwable>` that contains a possible write failure

But TBH I don't mind what's present as-is. Will let you two settle this if it's a blocker

Reply:
@wicknicks Since `encounteredErrors` can be modified concurrently, if we just return the number of errors to the caller, we can have the caller throw an exception, but we will lose the ability to include detailed error message. What do you think if we return an Optional of error string, and throw or no-op in the caller?

Reply:
yeah, this is a minor comment. ok to rename it also. any of @C0urante's suggestions are representative of what the method does (`maybeFail()` seems like the best)?